src/charm.py

```diff
@@ -566,7 +592,7 @@ def _bessd_pebble_layer(self) -> Layer:
                     self._bessd_service_name: {
                         "override": "replace",
                         "startup": "enabled",
-                        "command": f"/bin/bessd -f -grpc-url=0.0.0.0:{BESSD_PORT} -m 0",  # "-m 0" means that we are not using hugepages  # noqa: E501
+                        "command": f"/bin/bessd -f -grpc-url=0.0.0.0:{BESSD_PORT} {'-m 0' if not self.hugepages_is_enabled() else ''}",  # "-m 0" means that we are not using hugepages  # noqa: E501
```

Comment:
This gets hard to read, I'd be fine with a `_generate_bessd_startup_command` method that generates this command.